src/main/c/powerwaf_jni.c

```diff
@@ -1778,6 +1845,29 @@ static ddwaf_object _convert_checked(JNIEnv *env, jobject obj,
     return _pwinput_invalid;
 }
 
+static ddwaf_object _convert_buffer_checked(JNIEnv *env, jobject buffer)
+{
+    ddwaf_object result;
```

Comment:
you're converting java/C nulls to a null ddwaf_object, and then you convert then back C nulls before the call. The original code seems to have

```c
memcpy(&result, input_p, sizeof result);
```

in order to unify the byte_buffer and non byte_buffer variants.  But maybe this function still looks better returning `ddwaf_object *` instead and do:

```c
if (buffer == NULL) {
  return NULL;
}
void *input_p = JNI(GetDirectBufferAddress, buffer);
// error checking
return input_p;
```

Reply:
Done